runtime/startup_aux.c

```diff
@@ -168,14 +168,14 @@ int caml_startup_aux(int pooling)
   caml_verb_gc = 0x3F;
 #endif
   caml_parse_ocamlrunparam();
-  if (caml_cleanup_on_exit)
-    pooling = 1;
+  if (pooling)
+    caml_cleanup_on_exit = 1;
```

Comment:
I don't see the meaning in this change.

Previously, if the user set `OCAMLRUNPARAM=c`, then `caml_cleanup_on_exit` became 1, which enabled pooling.

Now, if pooling is enabled (via a call to `caml_startup_pooled`), it sets `caml_cleanup_on_exit` to 1, which later forces `caml_shutdown` on `exit`. But the user didn't ask for this! Leak detection mode (`caml_cleanup_on_exit`) isn't the same as ability to unload the runtime manually.

Reply:
I have removed this change, which stemmed from my misunderstanding. I also add a clarification of the difference between the two features in the documentation if you accept, to explain what you just wrote.